`undefined reference` errors during debug template compilation

### Tested versions

Reproducible in 4.3.dev (8ff8216)

### System information

Windows 11 - Godot 4.3.dev

### Issue description

When I try to compile the master branch with the `use_mingw=yes platform=windows dev_mode=yes target=template_debug` SCons options set, it fails at around 72% and prints the following info:

```
[ 72%] x86_64-w64-mingw32-g++ -o bin\godot.windows.template_debug.x86_64.exe -Wl,--subsystem,windows -static -Wl,--stack,8388608 -s platform\windows\godot_windows.windows.template_debug.x86_64.o platform\windows\crash_handler_windows.windows.template_debug.x86_64.o platform\windows\os_windows.windows.template_debug.x86_64.o platform\windows\display_server_windows.windows.template_debug.x86_64.o platform\windows\key_mapping_windows.windows.template_debug.x86_64.o platform\windows\joypad_windows.windows.template_debug.x86_64.o platform\windows\tts_windows.windows.template_debug.x86_64.o platform\windows\windows_terminal_logger.windows.template_debug.x86_64.o platform\windows\gl_manager_windows_native.windows.template_debug.x86_64.o platform\windows\gl_manager_windows_angle.windows.template_debug.x86_64.o platform\windows\wgl_detect_version.windows.template_debug.x86_64.o platform\windows\rendering_context_driver_vulkan_windows.windows.template_debug.x86_64.o platform\windows\godot_res.windows.template_debug.x86_64.o main\libmain.windows.template_debug.x86_64.a tests\libtests.windows.template_debug.x86_64.a modules\libmodules.windows.template_debug.x86_64.a modules\libmodule_ktx.windows.template_debug.x86_64.a modules\libmodule_msdfgen.windows.template_debug.x86_64.a modules\libmodule_theora.windows.template_debug.x86_64.a modules\libmodule_vorbis.windows.template_debug.x86_64.a modules\libmodule_basis_universal.windows.template_debug.x86_64.a modules\libmodule_bmp.windows.template_debug.x86_64.a modules\libmodule_camera.windows.template_debug.x86_64.a modules\libmodule_csg.windows.template_debug.x86_64.a modules\libmodule_dds.windows.template_debug.x86_64.a modules\libmodule_enet.windows.template_debug.x86_64.a modules\libmodule_freetype.windows.template_debug.x86_64.a modules\libmodule_gdscript.windows.template_debug.x86_64.a modules\libmodule_glslang.windows.template_debug.x86_64.a modules\libmodule_gltf.windows.template_debug.x86_64.a modules\libmodule_gridmap.windows.template_debug.x86_64.a modules\libmodule_hdr.windows.template_debug.x86_64.a modules\libmodule_jpg.windows.template_debug.x86_64.a modules\libmodule_jsonrpc.windows.template_debug.x86_64.a modules\libmodule_mbedtls.windows.template_debug.x86_64.a modules\libmodule_meshoptimizer.windows.template_debug.x86_64.a modules\libmodule_minimp3.windows.template_debug.x86_64.a modules\libmodule_mobile_vr.windows.template_debug.x86_64.a modules\libmodule_multiplayer.windows.template_debug.x86_64.a modules\libmodule_navigation.windows.template_debug.x86_64.a modules\libmodule_noise.windows.template_debug.x86_64.a modules\libmodule_ogg.windows.template_debug.x86_64.a modules\libmodule_openxr.windows.template_debug.x86_64.a modules\libmodule_raycast.windows.template_debug.x86_64.a modules\libmodule_regex.windows.template_debug.x86_64.a modules\libmodule_squish.windows.template_debug.x86_64.a modules\libmodule_svg.windows.template_debug.x86_64.a modules\libmodule_text_server_adv.windows.template_debug.x86_64.a modules\libmodule_tga.windows.template_debug.x86_64.a modules\libmodule_upnp.windows.template_debug.x86_64.a modules\libmodule_vhacd.windows.template_debug.x86_64.a modules\libmodule_webp.windows.template_debug.x86_64.a modules\libmodule_webrtc.windows.template_debug.x86_64.a modules\libmodule_websocket.windows.template_debug.x86_64.a modules\libmodule_webxr.windows.template_debug.x86_64.a modules\libmodule_zip.windows.template_debug.x86_64.a platform\libplatform.windows.template_debug.x86_64.a drivers\libdrivers.windows.template_debug.x86_64.a scene\libscene.windows.template_debug.x86_64.a servers\libservers.windows.template_debug.x86_64.a core\libcore.windows.template_debug.x86_64.a modules\text_server_adv\libharfbuzz_builtin.windows.template_debug.x86_64.a modules\text_server_adv\libgraphite_builtin.windows.template_debug.x86_64.a modules\text_server_adv\libicu_builtin.windows.template_debug.x86_64.a modules\freetype\libfreetype_builtin.windows.template_debug.x86_64.a modules\msdfgen\libmsdfgen_builtin.windows.template_debug.x86_64.a -lmingw32 -ldsound -lole32 -ld3d9 -lwinmm -lgdi32 -liphlpapi -lshlwapi -lwsock32 -lws2_32 -lkernel32 -loleaut32 -lsapi -ldinput8 -ldxguid -lksuser -limm32 -lbcrypt -lcrypt32 -lavrt -luuid -ldwmapi -ldwrite -lwbemuuid -lntdll -lpsapi -ldbghelp -lpsapi
=====
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tests\libtests.windows.template_debug.x86_64.a(test_main.windows.template_debug.x86_64.o):test_main.cpp:(.text$_ZN21GodotTestCaseListener13test_case_endERKN7doctest20CurrentTestCaseStatsE[_ZN21GodotTestCaseListener13test_case_endERKN7doctest20CurrentTestCaseStatsE]+0xc): undefined reference to `EditorSettings::get_singleton()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tests\libtests.windows.template_debug.x86_64.a(test_main.windows.template_debug.x86_64.o):test_main.cpp:(.text$_ZN21GodotTestCaseListener13test_case_endERKN7doctest20CurrentTestCaseStatsE[_ZN21GodotTestCaseListener13test_case_endERKN7doctest20CurrentTestCaseStatsE]+0x16): undefined reference to `EditorSettings::destroy()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tests\libtests.windows.template_debug.x86_64.a(test_main.windows.template_debug.x86_64.o):test_main.cpp:(.text$_ZN21GodotTestCaseListener15test_case_startERKN7doctest12TestCaseDataE[_ZN21GodotTestCaseListener15test_case_startERKN7doctest12TestCaseDataE]+0x40e): undefined reference to `EditorPaths::create()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tests\libtests.windows.template_debug.x86_64.a(test_main.windows.template_debug.x86_64.o):test_main.cpp:(.text$_ZN21GodotTestCaseListener15test_case_startERKN7doctest12TestCaseDataE[_ZN21GodotTestCaseListener15test_case_startERKN7doctest12TestCaseDataE]+0x413): undefined reference to `EditorSettings::create()'
collect2.exe: error: ld returned 1 exit status

=====
scons: *** [bin\godot.windows.template_debug.x86_64.exe] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:54:41.634]
```

Compilation was performed as a clean build (after using the `-c` Scons flag to clean targets).

Everything boils down to:

- `undefined reference to 'EditorSettings::get_singleton()'`
- `undefined reference to 'EditorSettings::destroy()'`
- `undefined reference to 'EditorPaths::create()'`
- `undefined reference to 'EditorSettings::create()'`


### Steps to reproduce

- Try to build the source from the commit mentioned above with the following options set:
  - `use_mingw=yes platform=windows dev_mode=yes target=template_debug`


### Minimal reproduction project (MRP)

N/A

--- a/tests/test_main.cpp
+++ b/tests/test_main.cpp
@@ -30,8 +30,11 @@
 
 #include "test_main.h"
 
+#ifdef TOOLS_ENABLED
 #include "editor/editor_paths.h"
 #include "editor/editor_settings.h"
+#endif // TOOLS_ENABLED
+
 #include "tests/core/config/test_project_settings.h"
 #include "tests/core/input/test_input_event.h"
 #include "tests/core/input/test_input_event_key.h"
@@ -270,11 +273,13 @@ struct GodotTestCaseListener : public doctest::IReporter {
 				SceneTree::get_singleton()->get_root()->set_embedding_subwindows(true);
 			}
 
+#ifdef TOOLS_ENABLED
 			if (name.find("[Editor]") != -1) {
 				Engine::get_singleton()->set_editor_hint(true);
 				EditorPaths::create();
 				EditorSettings::create();
 			}
+#endif // TOOLS_ENABLED
 
 			return;
 		}
@@ -298,9 +303,11 @@ struct GodotTestCaseListener : public doctest::IReporter {
 	}
 
 	void test_case_end(const doctest::CurrentTestCaseStats &) override {
+#ifdef TOOLS_ENABLED
 		if (EditorSettings::get_singleton()) {
 			EditorSettings::destroy();
 		}
+#endif // TOOLS_ENABLED
 
 		Engine::get_singleton()->set_editor_hint(false);
 

